cognite/neat/rules/_validation.py

```diff
@@ -1,10 +1,23 @@
 import re
+from typing import overload
 import warnings
 
 from cognite.neat.rules import _exceptions
 from cognite.neat.rules.models import TransformationRules, data_model_name_compliance_regex
 
 
+@overload
+def are_entity_names_dms_compliant(
+    transformation_rules: TransformationRules, return_report: bool = False
+) -> tuple[bool, list[dict]]:
+    ...
+
+
+@overload
+def are_entity_names_dms_compliant(transformation_rules: TransformationRules, return_report: bool = False) -> bool:
+    ...
```

Comment:
The same signature with different return value. 

```suggestion
@overload
def are_entity_names_dms_compliant(
    transformation_rules: TransformationRules, return_report: Literal[True]
) -> tuple[bool, list[dict]]:
    ...


@overload
def are_entity_names_dms_compliant(transformation_rules: TransformationRules, return_report: Literal[False] = False) -> bool:
    ...
```

I know this a strange syntax with `Literal[False]` but it is just the way it is.